includes/data-stores/class-wc-admin-reports-orders-data-store.php

```diff
@@ -497,23 +500,23 @@ protected static function get_num_items_sold( $order ) {
 	 * @return bool
 	 */
 	protected static function is_returning_customer( $order ) {
-		$customer_id = $order->get_user_id();
+		global $wpdb;
+		$customer_id        = WC_Admin_Reports_Customers_Data_Store::get_customer_id( $order->get_user_id() );
+		$orders_stats_table = $wpdb->prefix . self::TABLE_NAME;
 
-		if ( 0 === $customer_id ) {
+		if ( ! $customer_id ) {
 			return false;
 		}
 
-		$customer_orders = get_posts(
-			array(
-				'meta_key'    => '_customer_user', // WPCS: slow query ok.
-				'meta_value'  => $customer_id, // WPCS: slow query ok.
-				'post_type'   => 'shop_order',
-				'post_status' => array( 'wc-on-hold', 'wc-processing', 'wc-completed' ),
-				'numberposts' => 2,
+		$customer_orders = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT COUNT(*) FROM ${orders_stats_table} WHERE customer_id = %d AND date_created < %s",
+				$customer_id,
+				$order->get_date_created()->getTimestamp()
```

Comment:
>If you request a period where a customer placed their first order and subsequent orders, this person is tracked as new since they became a new customer during this period.

If I placed an order in November

<img width="311" alt="screen shot 2018-12-31 at 5 58 56 pm" src="https://user-images.githubusercontent.com/1922453/50554770-d0d3c000-0d25-11e9-8781-2bb110bcf283.png">

And place an order in December, then Orders Report for this month (December 1 -31st), should show me as a returning customer:

<img width="493" alt="screen shot 2018-12-31 at 5 53 05 pm" src="https://user-images.githubusercontent.com/1922453/50554785-09739980-0d26-11e9-8fc6-3c8e988b2383.png">

Maybe something wrong with this query?

Reply:
Maybe I'm looking at it wrong, but `AND date_created < %s` should use the `after` query parameter, not the order timestamp so new/returning would be relative to the time window in question.

Reply:
@psealock `is_returning_customer()` is actually a static method run during order updates and report building and will not have access to `before` or `after` query params.

The `is_returning_customer` column is stored as a bool in the order lookup table. 
 This function will check if a user with the same ID made an order before the date of the order in question and if so, marks this one as a returning customer.

Do you think there needs to be any changes to the logic here?

Reply:
If you have an order with a customer ID filled in from a previous period (as in your example where a customer ID is filled on an order in November and you're checking Dec 1-31st where an order is present with the same customer ID) then this person should be marked as returning.

If you have orders where the customer ID is filled in for those conditions, then the logic may be buggy here.